File: src/main/java/com/schoolSys/schooolSys/circulaire/Circulaire.java
package com.schoolSys.schooolSys.circulaire;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLRestriction;

@Entity
@Table(name = "circulaires")
@SQLDelete(sql = "UPDATE circulaires SET deleted = true, deleted_at = NOW() WHERE id = ?")
@SQLRestriction("deleted = false")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Circulaire {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 255)
    private String titre;

    @Column(nullable = false, length = 50)
    private String type;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String contenu;

    @Column(name = "date_creation", nullable = false)
    @Builder.Default
    private LocalDateTime dateCreation = LocalDateTime.now();

    @Column(name = "date_publication")
    private LocalDateTime datePublication;

    @Column(nullable = false, length = 30)
    @Builder.Default
    private String statut = "Brouillon";

    @Column(nullable = false, length = 255)
    private String auteur;

    @Column(nullable = false, length = 50)
    @Builder.Default
    private String cible = "Tous";

    @Column(name = "piece_jointe", length = 500)
    private String pieceJointe;

    @Column
    @Builder.Default
    private Boolean important = false;

    @Column(name = "created_at", nullable = false, updatable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at", nullable = false)
    @Builder.Default
    private LocalDateTime updatedAt = LocalDateTime.now();

    @Column(nullable = false)
    @Builder.Default
    private Boolean deleted = false;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;
}

File: school-system-front/src/types/circulaire.ts
export type TypeCirculaire = "Information" | "Règlement" | "Événement" | "Urgent" | "Pédagogique";

export type StatutCirculaire = "Brouillon" | "Publiée" | "Archivée";

export type CibleCirculaire = "Tous" | "Enseignants" | "Parents" | "Élèves" | "Administration";

export type Circulaire = {
  id: string;
  titre: string;
  type: TypeCirculaire;
  contenu: string;
  dateCreation: string;
  datePublication: string | null;
  statut: StatutCirculaire;
  auteur: string;
  cible: CibleCirculaire;
  pieceJointe: string | null;
  important: boolean;
};

export const TYPES_CIRCULAIRE: TypeCirculaire[] = [
  "Information",
  "Règlement",
  "Événement",
  "Urgent",
  "Pédagogique",
];

export const STATUTS_CIRCULAIRE: StatutCirculaire[] = [
  "Brouillon",
  "Publiée",
  "Archivée",
];

export const CIBLES_CIRCULAIRE: CibleCirculaire[] = [
  "Tous",
  "Enseignants",
  "Parents",
  "Élèves",
  "Administration",
];
